ext/BijectorsTapirExt.jl

```diff
@@ -0,0 +1,13 @@
+module BijectorsTapirExt
+
+if isdefined(Base, :get_extension)
+    using Tapir: @from_rrule, MinimalCtx, Tapir
+    using Bijectors: find_alpha
+else
+    using ..Tapir: @from_rrule, MinimalCtx, Tapir
+    using ..Bijectors: find_alpha
+end
+
+@from_rrule MinimalCtx Tuple{typeof(find_alpha),Float64,Float64,Float64}
```

Comment:
Does this imply that Tapir will error for e.g. `Float32` inputs?

Reply:
Yes -- Tapir.jl doesn't _really_ support `Float32` in general at the minute. This is because we've not set up testing infrastructure for it, and therefore dont want to make any promises about it.

In this particular instance, the rule won't be hit if `Float32`s are used, Tapir.jl will recurse into the definition, and eventually hit the bitcast that meant that this rule is necessary in the `Float64` case.

Reply:
I see. I wonder if it would be worth it (and possible) to import (or write, it's very simple) the rule in a more generic way such that Bijectors is already ready once Tapir supports more generic number types?

Reply:
I could definitely just make it applicable to `Union{Float32, Float64}` to future proof it. Is that the kind of thing you have in mind?

Reply:
Yes, something like that. The `rrule` is defined for `Real` but I guess `Float32` and `Float64` are the most relevant types; possibly `Float16` could be interesting as well. I assume Tapir might never be expected to work with more generic types such as `Real` or `AbstractFloat`?

Reply:
Hmm, from a quick look at the `find_alpha` occurrences in the code it seems that at least the last argument could easily be something more generic such as `Int` since it's directly taken from the user input. For the first argument it's unlikely but in principle also possible. The second argument should always be a floating point number (since its computation involves `log1pexp`).

Reply:
> Yes, something like that. The rrule is defined for Real but I guess Float32 and Float64 are the most relevant types; possibly Float16 could be interesting as well.

Good point -- I should just extend it to `Base.IEEEFloat`.

> I assume Tapir might never be expected to work with more generic types such as Real or AbstractFloat?

Short answer: a concrete subtype of `AbstractFloat` is unlikely to work. A concrete subtype of `Real` is unlikely to work if it's differentiable (e.g. if it's `Int`-like or `Bool`-like then it might work, because it's not a differentiable type, and we can largely exclude it from computation).

Slightly longer answer: in order to make a new concrete subtype of `AbstractFloat` work, Tapir.jl would either have to
1. differentiate through the primal code that Julia generates to run the function on this new type, or
2. specify a custom `tangent_type` and ensure there are rules for everything.

1 isn't likely to work because, much of the time, you'll wind up with quite a bit of `bitcast`-ing going on to implement things (i.e. so that you can do bit-twiddling stuff to make computations efficient), and Tapir.jl definitely can't differentiate through that kind of thing, and will (should) complain loudly if you try. This is to avoid silently giving the wrong answer.

2 requires work.

So the longer answer is effectively: it could be made to work, but would require effort.

> Hmm, from a quick look at the find_alpha occurrences in the code it seems that at least the last argument could easily be something more generic such as Int since it's directly taken from the user input. For the first argument it's unlikely but in principle also possible.

Okay -- I can definitely add support for this. It'll just be a few more lines of code in the extension, and some extra tests somewhere.

Reply:
Okay. Changes included.

@devmotion I've added support for `Integer` final arguments, and subtypes of `IEEEFloat` as inputs. I'll add support for Integer first arguments at some point in the future -- I need to extend the scope of Tapir's `ChainRules.rrule` conversion macro to support automatically dropped gradients w.r.t. isbits non-differentiable types (like integers), so this should become more straightforward in the future.